webapp/src/components/BuyEstatePage/BuyEstatePage.js

```diff
@@ -0,0 +1,248 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import { Loader, Container, Header, Grid, Message } from 'semantic-ui-react'
+
+import { locations } from 'locations'
+import Estate from 'components/Estate'
+import EstateModal from 'components/EstateDetailPage/EditEstateMetadata/EstateModal'
+// import TxStatus from 'components/TxStatus'
+import EstateName from 'components/EstateName'
+import Mana from 'components/Mana'
+import {
+  walletType,
+  authorizationType,
+  publicationType
+} from 'components/types'
+import { t, T } from '@dapps/modules/translation/utils'
+import { isLegacyPublication } from 'modules/publication/utils'
+import { formatMana } from 'lib/utils'
+
+import './BuyEstatePage.css'
+
+export default class BuyEstatePage extends React.PureComponent {
+  static propTypes = {
+    id: PropTypes.string.isRequired,
+    wallet: walletType,
+    authorization: authorizationType,
+    publication: publicationType,
+    isDisabled: PropTypes.bool.isRequired,
+    isLoading: PropTypes.bool.isRequired,
+    isConnected: PropTypes.bool.isRequired,
+    onConfirm: PropTypes.func.isRequired,
+    onCancel: PropTypes.func.isRequired
+  }
+
+  static defaultProps = {
+    publication: { price: 0 }
```

Comment:
we should let the UI handle an undefined/null publication rather than mocking a default one